examples/platforms/nrf528xx/src/radio.c

```diff
@@ -1118,3 +1158,52 @@ void otPlatRadioSetMacFrameCounter(otInstance *aInstance, uint32_t aMacFrameCoun
     CRITICAL_REGION_EXIT();
 }
 #endif // OPENTHREAD_CONFIG_THREAD_VERSION >= OT_THREAD_VERSION_1_2
+
+#if OPENTHREAD_CONFIG_MAC_CSL_RECEIVER_ENABLE
+static void updateIeData(otInstance *aInstance, const uint8_t *aShortAddr, const uint8_t *aExtAddr)
+{
+    int8_t  offset = 0;
+    uint8_t ackIeData[OT_ACK_IE_MAX_SIZE];
+
+    if (sCslPeriod > 0)
+    {
+        memcpy(ackIeData, sCslIeHeader, OT_IE_HEADER_SIZE);
+        offset += OT_IE_HEADER_SIZE + OT_CSL_IE_SIZE; // reserve space for CSL IE
+    }
+
+    if (offset > 0)
+    {
+        nrf_802154_ack_data_set(aShortAddr, false, ackIeData, offset, NRF_802154_ACK_DATA_IE);
+        nrf_802154_ack_data_set(aExtAddr, true, ackIeData, offset, NRF_802154_ACK_DATA_IE);
+    }
+    else
+    {
+        nrf_802154_ack_data_clear(aShortAddr, false, NRF_802154_ACK_DATA_IE);
+        nrf_802154_ack_data_clear(aExtAddr, true, NRF_802154_ACK_DATA_IE);
+    }
+}
+
+otError otPlatRadioEnableCsl(otInstance *aInstance, uint32_t aCslPeriod, const otExtAddress *aExtAddr)
+{
+    otError error = OT_ERROR_NONE;
+    uint8_t parentExtAddr[OT_EXT_ADDRESS_SIZE];
+
+    sCslPeriod = aCslPeriod;
+
+    for (uint32_t i = 0; i < sizeof(parentExtAddr); i++)
+    {
+        parentExtAddr[i] = aExtAddr->m8[sizeof(parentExtAddr) - i - 1];
+    }
+
+    updateIeData(aInstance, sParentShortAddress, parentExtAddr);
+
+    return error;
+}
+
+void otPlatRadioUpdateCslSampleTime(otInstance *aInstance, uint32_t aCslSampleTime)
```

Comment:
should aCslSampleTime use 10 symbols unit as the period?

Reply:
Make sense. This would also cause a change in simulation radio. Shall we do it in another PR as an enhancement?